dnf/cli/commands/__init__.py

```diff
@@ -913,39 +925,66 @@ def str2tid(s):
         tids = set()
         for t in self.opts.tid:
             if '..' in t:
-                btid, etid = t.split('..', 2)
-                btid = str2tid(btid)
-                etid = str2tid(etid)
+                try:
+                    btid, etid = t.split('..', 2)
+                except ValueError:
+                    logger.critical(
+                        _("Invalid transaction ID range definition '{}'.\n"
+                          "Use '<transaction-id>..<transaction-id>'."
+                          ).format(t))
+                    raise dnf.cli.CliError
+                cant_convert_msg = "Can't convert '{}' to transaction ID.\n" \
+                                   "Use '<integer>' or 'last' or 'last-<positive-integer>'."
+                try:
+                    btid = str2tid(btid)
+                except ValueError:
+                    logger.critical(_(cant_convert_msg).format(btid))
```

Comment:
I can't do that. `str2tid` is called on more places. And exception handling is different